initfiles/sbin/spark-hpcc.sh.in

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+
+exec 3>&2 2>@LOG_PATH@/spark-hpcc.debug
+set -x
+
+_command=${1:-start}
+_component=${2:-myspark-hpcc}
+
+LOCAL_IP="$(@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -listall -ip localhost | awk -F "," 'NR==1{print $3;}')"
+@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -listall2 -ip localhost -t thor | grep "ThorSlaveProcess" 2>&1 1>/dev/null
+_IS_THORSLAVE=$?
+# Always use the first record returned by `configgen -t esp` as master
+MASTER_IP="$(@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -listall -t esp | awk -F "," 'NR==1{print $3;}')"
+MASTER_URL="spark://${MASTER_IP}:7077"
+
+
+export SPARK_LOG_DIR="@LOG_PATH@/${_component}"
+export SPARK_PID_DIR="@PID_PATH@"
+if [[ "${LOCAL_IP}" == "${MASTER_IP}" ]]; then
+    @INSTALL_DIR@/externals/spark-hadoop/sbin/${_command}-master.sh
+fi
+
+export SPARK_WORKER_DIR="@RUNTIME_PATH@/${_component}"
+if [[ ${_IS_THORSLAVE} == 0 ]]; then
+    @INSTALL_DIR@/externals/spark-hadoop/sbin/${_command}-slave.sh ${MASTER_URL}
```

Comment:
is spark-hadoop a folder we create, or is it created by the spark install process?

Reply:
It's a part of the spark install process.